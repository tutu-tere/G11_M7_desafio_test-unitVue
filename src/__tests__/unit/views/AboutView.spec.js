import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import AboutView from '@/views/AboutView.vue'
import { createRouter, createWebHistory } from 'vue-router'
import App from '@/App.vue'

describe('AboutView.vue', () => {
  it('Contrasta Snapshot del HTML', () => {
    const wrapper = mount(AboutView)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Se muestra la vista About al llegar a través de la ruta', async () => {
    /**
     * Definimos configuración base para la prueba
     */
    const routerPrueba = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: '/about',
          name: 'about',
          component: AboutView
        }
      ]
    })

    const wrapper = mount(App, {
      global: {
        plugins: [routerPrueba]
      }
    })

    routerPrueba.push({ name: 'about' })

    await routerPrueba.isReady()

    expect(wrapper.findComponent(AboutView).exists()).toBeTruthy()
  })
})
