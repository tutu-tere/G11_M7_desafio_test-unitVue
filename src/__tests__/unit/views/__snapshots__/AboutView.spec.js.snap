// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutView.vue > Contrasta Snapshot del HTML 1`] = `
"<div class="about">
  <h1>This is an about page</h1>
</div>"
`;
